packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -19,6 +24,11 @@ import useBlockDisplayInformation from '../use-block-display-information';
 import BlockTitle from '../block-title';
 import ListViewExpander from './expander';
 import { useBlockLock } from '../block-lock';
+import { store as blockEditorStore } from '../../store';
+
+// For this list of hard-coded blocks, the block content will be used as the button label.
+// If no content exists, then the block's title will be used as a fallback.
+const CONTENT_LABEL_BLOCKS = [ 'core/heading', 'core/paragraph' ];
```

Comment:
It might be a bit much including the `paragraph` here, so we might want to pare this back to just the `heading` block for now 🤔

Reply:
Let's start with heading, I think paragraph might be worth treating differently:

<img width="1077" alt="image" src="https://user-images.githubusercontent.com/548849/175005743-2324890c-43b6-4ee8-b307-05f125147f60.png">

Reply:
I've updated so that it only affects the Heading block for now. I'm happy to explore that Paragraph idea in a follow-up if no-one beats me to it, looks like it could be a good way to handle it!